x-pack/plugins/enterprise_search/public/applications/app_search/components/documents/search_experience/search_experience_content.test.tsx

```diff
@@ -0,0 +1,160 @@
+      ...searchState,
+      wasSearched: false,
+    });
+    const wrapper = shallow(<SearchExperienceContent />);
+    expect(wrapper.isEmptyRender()).toBe(true);
+  });
+
+  it('renders results if a search was performend and there are more than 0 totalResults', () => {
+    setMockSearchContextState({
+      ...searchState,
+      wasSearched: true,
+      totalResults: 10,
+    });
+    const wrapper = shallow(<SearchExperienceContent />);
+    expect(wrapper.find('[data-test-subj="documentsSearchResults"]').length).toBe(1);
+  });
+
+  it('renders a no results message if a non-empty search was performed and there are no results', () => {
+    setMockSearchContextState({
+      ...searchState,
+      resultSearchTerm: 'searchTerm',
+      wasSearched: true,
+      totalResults: 0,
+    });
+    const wrapper = shallow(<SearchExperienceContent />);
+    expect(wrapper.find('[data-test-subj="documentsSearchResults"]').length).toBe(0);
+    expect(wrapper.find('[data-test-subj="documentsSearchNoResults"]').length).toBe(1);
+  });
+
+  describe('when an empty search was performed and there are no results, meaning there are no documents indexed', () => {
+    beforeEach(() => {
+      setMockSearchContextState({
+        ...searchState,
+        resultSearchTerm: '',
+        wasSearched: true,
+        totalResults: 0,
+      });
+    });
+    it('renders a no documents message', () => {
+      const wrapper = shallow(<SearchExperienceContent />);
+      expect(wrapper.find('[data-test-subj="documentsSearchResults"]').length).toBe(0);
+      expect(wrapper.find('[data-test-subj="documentsSearchNoDocuments"]').length).toBe(1);
+    });
+
+    it('will include a button to index new documents', () => {
+      const wrapper = shallow(<SearchExperienceContent />);
+      const noDocumentsErrorActions: any = wrapper
+        .find('[data-test-subj="documentsSearchNoDocuments"]')
+        .prop('actions');
+
+      expect(
+        shallow(noDocumentsErrorActions).find('[data-test-subj="IndexDocumentsButton"]').length
+      ).toBe(1);
+    });
+
+    it('will include a button to documentation if this is a meta engine', () => {
+      setMockValues({
+        ...values,
+        isMetaEngine: true,
+      });
+
+      const wrapper = shallow(<SearchExperienceContent />);
+
+      const noDocumentsErrorActions: any = wrapper
+        .find('[data-test-subj="documentsSearchNoDocuments"]')
+        .prop('actions');
+
+      expect(
+        shallow(noDocumentsErrorActions).find('[data-test-subj="IndexDocumentsButton"]').length
+      ).toBe(0);
+      expect(
+        shallow(noDocumentsErrorActions).find('[data-test-subj="documentsSearchDocsLink"]').length
+      ).toBe(1);
+    });
+
+    it('will include a button to documentation if the user cannot manage documents', () => {
+      setMockValues({
+        ...values,
+        myRole: { canManageEngineDocuments: false },
+      });
+
+      const wrapper = shallow(<SearchExperienceContent />);
```

Comment:
IMO, for the 3 tests in these blocks, feel free to just use `mount()` instead of `shallow()` so you don't have to deal with re-diving into `noDocumentsErrorActions` repeatedly

Reply:
6c892570c9945950a2dd167650842b642969bc01